src/naf-dialog-adapter.js

```diff
@@ -192,161 +188,122 @@ export default class DialogAdapter {
-   * Firefox doesn't support hot updating ICE servers through mediasoup's updateIceServers (which
-   * internally uses setConfiguration):
-   * https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/setConfiguration
-   * So we recreate the transports/consumer/producers through reconnect to force setting
-   * new ICE servers for both send and receive transports.
-   */
+  async iceRestart(transport) {
+    // Force an ICE restart to gather new candidates and trigger a reconnection
+    this.emitRTCEvent(
+      "log",
+      "RTC",
+      () => `Restarting ${transport.id === this._sendTransport.id ? "send" : "receive"} transport ICE`
+    );
+    const iceParameters = await this._protoo.request("restartIce", { transportId: transport.id });
+    await transport.restartIce({ iceParameters });
+  }
+
+  async recreateSendTransport(iceServers) {
+    this.emitRTCEvent("log", "RTC", () => `Recreating send transport ICE`);
+    await this.closeSendTransport();
+    await this.createSendTransport(iceServers);
+  }
 
   /**
    * Restart ICE in the underlying send peerconnection.
    */
   async restartSendICE(force = true) {
-    if (!this._sendTransport?._closed) {
-      // Renegotiate ICE in case ICE state is "failed".
-      // Chrome doesn't seem to automatically reconnect after a "disconnected" so we renegotiate in that case.
-      // We also try to renegotiate in case we got stuck in a new state after an ICE failure.
-      if (
-        force ||
-        this._sendTransport.connectionState === "failed" ||
-        (!isFirefox && this._sendTransport.connectionState === "disconnected") ||
-        (this._sendTransport.connectionState === "new" && this._lastSendConnectionState === "failed")
-      ) {
-        this.emitRTCEvent("log", "RTC", () => `Restarting send transport ICE`);
-        if (this._protoo.connected) {
-          if (isFirefox) {
-            this.reconnect();
-          } else if (!this._sendTransport.closed) {
-            const { host, port, turn } = await window.APP.hubChannel.getHost();
-            const iceServers = this.getIceServers(host, port, turn);
-            await this._sendTransport.updateIceServers({ iceServers });
-            const iceParameters = await this._protoo.request("restartIce", { transportId: this._sendTransport.id });
-            await this._sendTransport.restartIce({ iceParameters });
-          }
+    // Do not restart ICE if Signaling is disconnected. We are not in the meeting room if that's the case.
+    if (this._closed) {
+      return;
+    }
+
+    try {
+      if (!this._sendTransport?._closed) {
+        // Renegotiate ICE in case ICE state is "failed".
+        // We also try to renegotiate in case we got stuck in a new state after an ICE failure.
+        if (
+          force ||
+          this._sendTransport.connectionState === "failed" ||
+          (this._sendTransport.connectionState === "new" && this._lastSendConnectionState === "failed")
```

Comment:
Not too sure what state this is checking for

Reply:
That's just a state I've seen a couple of time happening. I can just remove it and keep it simple until we have more insights about WebebRTC errors.